common/djangoapps/enrollment/views.py

```diff
@@ -598,8 +598,8 @@ def post(self, request):
             enterprise_course_consent = request.data.get('enterprise_course_consent')
             explicit_linked_enterprise = request.data.get('linked_enterprise_customer')
             if (enterprise_course_consent or explicit_linked_enterprise) and has_api_key_permissions and enterprise_enabled():
-                enterprise_api_client = EnterpriseApiClient()
-                consent_client = ConsentApiClient()
+                enterprise_api_client = EnterpriseApiClient(user=request.user)
```

Comment:
I think you need to use the service user here.

Reply:
Service user.